### PR TITLE
Add basic documentation

### PR DESCRIPTION
**Pull request by lewtun**
# What does this PR do?

Adds basic documentation that can be rendered on hf.co via https://github.com/huggingface/optimum/pull/311

I've also included a small guide so contributors to this project can understand how the docs should be formatted.

Note: we should remove the bits that checkout the `documentation` branch before merging this PR. That's just used as a test that the approach works end to end (it does :))

Needs https://github.com/huggingface/moon-landing/pull/3521 to be merged before the docs and be viewed from this PR

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: .github/workflows/check_doc_build.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'huggingface/optimum-graphcore'
+          path: optimum-graphcore
+
+      - name: Make documentation
+        run: |
+          cd optimum-graphcore
+          git fetch origin
```
Comment (lewtun) [inline .github/workflows/check_doc_build.yml:26]: These 2 lines should be removed before merging

File: docs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# Need node14 to build doc HTML. Taken from https://askubuntu.com/a/1113339
+RUN apt-get update && apt-get install -y \
+    git \
+    curl \
+    gnupg
+RUN curl -sL https://deb.nodesource.com/setup_14.x  | bash -
+RUN apt-get -y install nodejs
+RUN npm install
+
+RUN git clone -b documentation https://github.com/huggingface/optimum-graphcore.git
```
Comment (lewtun) [inline docs/Dockerfile:12]: Replace with the following before merging:

```suggestion
RUN git clone https://github.com/huggingface/optimum-graphcore.git
```

File: .github/workflows/check_doc_build.yml (hunk 1 above)
Comment (lewtun) [inline .github/workflows/check_doc_build.yml:26]: Remove these 2 lines before merging

File: docs/Dockerfile (hunk 2 above)
Comment (lewtun) [inline docs/Dockerfile:12]: Replace with this before merging:
```suggestion
RUN git clone https://github.com/huggingface/optimum-graphcore.git
```

File: optimum/graphcore/version.py [hunk 3]
```diff
@@ -12,4 +12,4 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-__version__ = "0.3.1.dev"
+__version__ = "0.3.1.dev0"
```
Comment (lewtun) [inline optimum/graphcore/version.py:15]: This is needed to distinguish builds on `main` versus official release branches

File: tests/test_trainer.py [hunk 4]
```diff
@@ -27,15 +27,15 @@
 
 import numpy as np
 
-from huggingface_hub import Repository, delete_repo, login
```
Comment (lewtun) [inline tests/test_trainer.py:30]: `login` is deprecated, along with using username / password. Refactored to use a dummy token instead

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+## Writing documenation examples
+
+The syntax for Example docstrings can look as follows:
+
+```
+    Example:
+
+    ```python
+    >>> from transformers import Wav2Vec2Processor, Wav2Vec2ForCTC
```
Comment (michaelbenayoun) [inline docs/README.md:270]: Maybe put some `optimum-graphcore` code example here?

**Review by michaelbenayoun — APPROVED** [review]
Thanks a lot for taking care of this!

File: docs/README.md (hunk 5 above)
Comment (lewtun) [reply docs/README.md:270]: Yeah good idea - I'll do that in a separate PR to first get the whole docs built in `optimum` :)

**Pull request merged**